tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolateEvent.cpp

```diff
@@ -0,0 +1,195 @@
+  using phase_dependent_action_list = tmpl::list<Parallel::PhaseActions<
+      typename Metavariables::Phase, Metavariables::Phase::Initialization,
+      tmpl::list<intrp::Actions::InitializeInterpolator>>>;
+  using initial_databox = db::compute_databox_type<
+      typename ::intrp::Actions::InitializeInterpolator::
+          template return_tag_list<Metavariables>>;
+};
+
+template <typename Metavariables>
+struct mock_element {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndex<Metavariables::domain_dim>;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using add_options_to_databox = Parallel::AddNoOptionsToDataBox;
+  using phase_dependent_action_list =
+      tmpl::list<Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Initialization,
+                                        tmpl::list<>>>;
+  using initial_databox = db::compute_databox_type<db::AddSimpleTags<>>;
+};
+
+struct MockMetavariables {
+  struct InterpolatorTargetA {
+    using vars_to_interpolate_to_target = tmpl::list<Tags::Lapse>;
+  };
+  using temporal_id = ::Tags::TimeId;
+  static constexpr size_t domain_dim = 3;
+  using interpolator_source_vars = tmpl::list<Tags::Lapse>;
+  using interpolation_target_tags = tmpl::list<InterpolatorTargetA>;
+
+  using component_list = tmpl::list<mock_interpolator<MockMetavariables>,
+                                    mock_element<MockMetavariables>>;
+  using const_global_cache_tag_list = tmpl::list<>;
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+SPECTRE_TEST_CASE("Unit.NumericalAlgorithms.Interpolator.InterpolateEvent",
+                  "[Unit]") {
+  using metavars = MockMetavariables;
+  const ElementId<metavars::domain_dim> element_id(2);
+  const ElementIndex<metavars::domain_dim> array_index(element_id);
+
+  using interp_component = mock_interpolator<metavars>;
+  using elem_component = mock_element<metavars>;
+  ActionTesting::MockRuntimeSystem<metavars> runner{{}};
+  runner.set_phase(metavars::Phase::Initialization);
+  ActionTesting::emplace_component<interp_component>(&runner, 0);
+  ActionTesting::next_action<interp_component>(make_not_null(&runner), 0);
+  ActionTesting::emplace_component<elem_component>(&runner, array_index);
+  ActionTesting::next_action<elem_component>(make_not_null(&runner),
+                                             array_index);
+  runner.set_phase(metavars::Phase::Testing);
+
+  const Mesh<metavars::domain_dim> mesh(5, Spectral::Basis::Legendre,
+                                        Spectral::Quadrature::GaussLobatto);
+  const double observation_time = 2.0;
+  Variables<metavars::interpolator_source_vars> vars(
+      mesh.number_of_grid_points());
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+  std::iota(vars.data(), vars.data() + vars.size(), 1.0);
```

Comment:
Just an FYI, a lot of the `std::` algorithms are wrapped in `Utilities/Algorithm.hpp` and `Utilities/Numeric.hpp` (mirroring the STL) and allow the simpler `alg::iota(var, 1.0);`. I'm happy to merge as is, please let me know whether or not you want to change it :)

Reply:
Talked offline. Doesn't work since Variables doesn't have begin and end